src/poetry/core/packages/directory_dependency.py

```diff
@@ -46,11 +45,12 @@ def __init__(
 
         # Checking content to determine actions
         setup = self._full_path / "setup.py"
-        self._supports_poetry = PyProjectTOML(
-            self._full_path / "pyproject.toml"
-        ).is_poetry_project()
+        pyproject = PyProjectTOML(self._full_path / "pyproject.toml")
 
-        if not setup.exists() and not self._supports_poetry:
+        self._supports_pep517 = setup.exists() or pyproject.is_build_system_defined
```

Comment:
Just noticed on a quick glance: `is_poetry_project()` is a method. Should `is_build_system_defined` be a method, too? (for consistency)